Client/Frontend/Sync/SyncSettingsTableViewController.swift

```diff
@@ -391,23 +392,8 @@ extension SyncSettingsTableViewController {
         self.devices = devices
 
         if devices.count <= 0 {
-          // Technically we shouldn't be calling this function..
-          // If Desktop deletes an iOS device from the chain, we're already removed..
-          // We should be just updating our UI and calling
-          // `Preferences.Chromium.syncEnabled.value = false`
-          // But I don't have a TRUE way to tell if the current device
-          // was removed from the sync chain because `OnSelfDeviceInfoDeleted` has no callback.
-          // So instead, we call `removeAllObservers` which won't hurt.
-          // Calling `syncAPI.leaveSyncGroup() aka reset_sync` as of 1.34.x will crash the application.
-
-          // Commented out in 1.38.x to see if it helps users
-          // If it does, this will be removed in 1.39.x
-          // Reasoning: I suspect the server removes THIS device from the chain for some reason
-          // When that happens, we get `devices.count == 0` and we're removing observers and setting
-          // syncEnabled to false. What happens if we get 0 devices (including our own device), but the user is still on the chain??? This should not be possible afaik, because there has to be ONE device (our own) / this device at minimum.
-          // If this fixes it for users, it means someone changed something without telling us (same as 1.34.x).
-          //syncAPI.removeAllObservers()
-          //Preferences.Chromium.syncEnabled.value = false
+           syncAPI.removeAllObservers()
```

Comment:
Is it safe to remove the observers now ?

Reply:
No it was a temporary fix until another Brave-Core PR is merged by Alexey to fix the encryption error (causes the user to get stuck on the sync chain)